src/Amqp/WorkQueue/DeclarationManager.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Icecave\Overpass\Amqp\WorkQueue;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+
+/**
+ * @access private
+ */
+class DeclarationManager
+{
+    public function __construct(AMQPChannel $channel)
+    {
+        $this->channel       = $channel;
+        $this->requestQueues = [];
+    }
+
+    /**
+     * @return string
+     */
+    public function exchange()
+    {
+        if ($this->exchange) {
+            return $this->exchange;
+        }
+
+        $name = 'overpass/task';
+
+        $this
+            ->channel
+            ->exchange_declare(
+                $name,
+                'direct',
+                false, // passive,
+                false, // durable,
+                false  // auto delete
+            );
+
+        $this->exchange = $name;
+
+        return $this->exchange;
+    }
+
+    /**
+     * @param string $procedureName
+     *
+     * @return string
+     */
+    public function requestQueue($procedureName)
+    {
+        if (isset($this->requestQueues[$procedureName])) {
+            return $this->requestQueues[$procedureName];
+        }
+
+        list($queueName) = $this
+            ->channel
+            ->queue_declare(
+                'overpass/task/' . $procedureName,
+                false, // passive
+                false, // durable
```

Comment:
We want to use a durable queue here, which means that the queue persists between rabbit restarts.
Queues need to be durable in order to save messages to disk, which is desirable for our job queue.